### ISSUE TITLE
styleguide - custom icons with id defined with single quotes cause 500 error when loading style guide

### ISSUE THREAD

**Issue by enzedonline** [wagtail/wagtail#11895]
### Issue Summary

If a registered custom icon uses single quotes to define the `id`, the `all_icons` context variable doesn't recognise the icon name (i.e. the name attribute is empty). These icons behave as expected elsewhere in the admin interface.

When the styleguide template loops through the custom icons, the icon template tag is called with an empty name attribute causing the 500 error ([ref](https://github.com/wagtail/wagtail/blob/9dc38076ed9d12d63dc40258e6135b43d4054d96/wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html#L524)).
```django
{% icon name=item.name classname="w-w-8 w-h-8" %}
```

### Steps to Reproduce

Add a custom svg to your project with the id defined with single quotes:

```xml
<svg
   viewBox="0 0 1024 1024"
   id='icon-ban'
   xmlns="http://www.w3.org/2000/svg"
   xmlns:svg="http://www.w3.org/2000/svg">
   <path
     d="M 734.4,825 199,289.6 C 154.2,352.2 128,429 128,512 c 0,212 172,384 384,384 83,0 159.8,-26.2 222.4,-71 z M 825,734.4 C 869.8,671.8 896,595 896,512 896,300 724,128 512,128 429,128 352.2,154.2 289.6,199 Z M 0,512 a 512,512 0 1 1 1024,0 512,512 0 1 1 -1024,0 z"
     style="stroke-width:2" 
   />
</svg>
```

[Register the icon](https://docs.wagtail.org/en/stable/advanced_topics/icons.html#add-a-custom-icon) as per docs.

Browse the style guide - the error is thrown in the icon template tag due to missing icon name.

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: Yes

### Technical details

-   Python version: 3.1.2
-   Django version: 4.2.7
-   Wagtail version: 6.0.2
-   Browser version: Chrome 124

**Comment by laymonage** [comment]
Thank you, I can confirm that this issue exists. It's been around since 5.0 though, so we might only fix this for 6.1 and up.

**Comment by thibaudcolas** [comment]
Fixed! This will ship in Wagtail 6.2, and the upcoming 6.1.2 patch release.
